protobuf/lib/src/protobuf/generated_message.dart

```diff
@@ -223,6 +223,13 @@ abstract class GeneratedMessage {
           {TypeRegistry typeRegistry = const TypeRegistry.empty()}) =>
       _writeToProto3Json(_fieldSet, typeRegistry);
 
+  String toProto3JsonString(
```

Comment:
Additions in this PR will be inconsistent unless we want to do breaking changes.

Currently in master branch (and also in the latest release) we have these generators:

- writeToBuffer
- writeToCodedBufferWriter
- writeToJson
- writeToJsonMap
- toProto3Json

So the proto3 method is already inconsistent.

Without changing it, we can name the new methods `toProto3...`, or `writeToProto3...`. Either way it will be inconsistent.

Should we rename `toProto3Json`? @sigurdm

Reply:
Argh - I see - I introduced that inconsistency - not sure what is the better path forwards.
I guess if we want to be consistent we do:
```
@Deprecated('Use `writeToProto3JsonMap`');
toProto3Json() => writeToProto3JsonMap();

writeToProto3JsonMap();
writeToProto3Json();
```

If we prefer to avoid churn over consistency we keep it as `toProto3JsonString()`....

I guess it's up to your judgement.